components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -598,14 +616,71 @@ void RewardsServiceImpl::CreateRewardsWallet(
         observer.OnRewardsWalletUpdated();
       }
 
-      std::move(callback).Run(ledger::mojom::Result::LEDGER_OK);
+      std::move(callback).Run(result);
     };
 
     self->bat_ledger_->CreateRewardsWallet(
-        base::BindOnce(on_created, self, std::move(callback)));
+        country,
+        base::BindOnce(on_created, self, country, std::move(callback)));
   };
 
-  StartProcess(base::BindOnce(on_start, AsWeakPtr(), std::move(callback)));
+  StartProcess(
+      base::BindOnce(on_start, AsWeakPtr(), country, std::move(callback)));
+}
+
+std::string RewardsServiceImpl::GetCountryCode() const {
+  auto* prefs = profile_->GetPrefs();
+  std::string country = prefs->GetString(prefs::kDeclaredGeo);
+  if (!country.empty()) {
+    return country;
+  }
+
+  int32_t country_id = country_id_;
+  if (!country_id) {
+    country_id = country_codes::GetCountryIDFromPrefs(prefs);
+  }
+  if (country_id < 0) {
+    return "";
+  }
+  return {base::ToUpperASCII(static_cast<char>(country_id >> 8)),
+          base::ToUpperASCII(static_cast<char>(0xFF & country_id))};
+}
+
+void RewardsServiceImpl::GetAvailableCountries(
+    GetAvailableCountriesCallback callback) const {
+  static const std::vector<std::string> iso_countries = GetISOContries();
+
+  if (!Connected()) {
+    return DeferCallback(FROM_HERE, std::move(callback), iso_countries);
+  }
+
+  auto on_external_wallet = [](GetAvailableCountriesCallback callback,
+                               std::vector<std::string> countries,
+                               ledger::mojom::Result result,
+                               ledger::mojom::ExternalWalletPtr wallet) {
+    // If the user is currently linked to a bitFlyer wallet, then the only
+    // available countries are |kBitflyerCountries|. If the user is currently
+    // linked to any other external wallet provider, then remove
+    // |kBitflyerCountries| from the list of ISO countries.
+    if (wallet &&
+        wallet->status != ledger::mojom::WalletStatus::NOT_CONNECTED) {
+      if (wallet->type == ledger::constant::kWalletBitflyer) {
+        countries = std::vector<std::string>(kBitflyerCountries.begin(),
+                                             kBitflyerCountries.end());
+      } else {
+        auto removed =
```

Comment:
Maybe use `base::ranges::set_difference<>()`?
It requires sorted input ranges, but ISO countries could also be stored in a `std::set<std::string>`.

Reply:
We could also cache the result, so that we don't have to calculate the difference more than once.

Reply:
I decided to cache the non-bF result, but for simplicity left the vector implementation for now.